sdk/samples/iot_mosquitto/mosquitto_sample_connect.c

```diff
@@ -0,0 +1,215 @@
+      log_level = "DBUG";
+      break;
+    case MOSQ_LOG_SUBSCRIBE:
+      log_level = "SUB ";
+      break;
+    case MOSQ_LOG_UNSUBSCRIBE:
+      log_level = "USUB";
+      break;
+    case MOSQ_LOG_WEBSOCKETS:
+      log_level = "WSCK";
+      break;
+    default:
+      log_level = "UNKN";
+  }
+  printf("MOSQ [%s] %s\n", log_level, str);
+}
+
+int main(int argc, char *argv[])
+{
+  (void)argc; (void)argv;
+
+  struct mosquitto *mosq;
+  int rc;
+
+  /* Required before calling other mosquitto functions */
+  mosquitto_lib_init();
+
+  /* Create a new client instance.
+   * id = NULL -> ask the broker to generate a client id for us
+   * clean session = true -> the broker should remove old sessions when we connect
+   * obj = NULL -> we aren't passing any of our private data for callbacks
+   */
+  mosq = mosquitto_new(MOSQUITTO_IOT_HUB_DEVICENAME, true, NULL);
+  if(mosq == NULL){
+    fprintf(stderr, "Error: Out of memory.\n");
+    return 1;
+  }
+
+  /* Configure callbacks. This should be done before connecting ideally. */
+  mosquitto_log_callback_set(mosq, on_log);
+
+  mosquitto_connect_callback_set(mosq, on_connect);
+  mosquitto_disconnect_callback_set(mosq, on_disconnect);
+  mosquitto_publish_callback_set(mosq, on_publish);
+  mosquitto_subscribe_callback_set(mosq, on_subscribe);
+  mosquitto_unsubscribe_callback_set(mosq, on_unsubscribe);
+  
+  rc = mosquitto_tls_set(
+    mosq,
+    MOSQUITTO_X509_TRUST_PEM_FILE_PATH,
+    NULL, //"S:\\cert\\RootCAs",
+    MOSQUITTO_X509_CERT_ONLY_PEM_FILE_PATH,
+    MOSQUITTO_X509_KEY_PEM_FILE_PATH,
+    NULL);
+  if (rc != MOSQ_ERR_SUCCESS)
+  {
+    mosquitto_destroy(mosq);
+    fprintf(stderr, "TLS Config Error: %s\n", mosquitto_strerror(rc));
+    return 1;
+  }
+
+  rc = mosquitto_username_pw_set(mosq, MOSQUITTO_USERNAME, "");
+  if (rc != MOSQ_ERR_SUCCESS)
+  {
+    mosquitto_destroy(mosq);
+    fprintf(stderr, "User/pass Config Error: %s\n", mosquitto_strerror(rc));
+    return 1;
+  }
+
+  rc = mosquitto_connect_async(mosq, MOSQUITTO_IOT_HUB_HOSTNAME, 8883, 60);
+  if(rc != MOSQ_ERR_SUCCESS)
+  {
+    mosquitto_destroy(mosq);
+    fprintf(stderr, "Error: %s\n", mosquitto_strerror(rc));
+    return 1;
+  }
+
+  // Mosquitto BUG: sleep required when connect async used on Windows.
+  //Sleep(500);
+
+  /* Run the network loop in a background thread, this call returns quickly. */
+  rc = mosquitto_loop_start(mosq);
+  if(rc != MOSQ_ERR_SUCCESS)
+  {
+    mosquitto_destroy(mosq);
+    fprintf(stderr, "Error: %s\n", mosquitto_strerror(rc));
+    return 1;
+  }
+
+  /* At this point the client is connected to the network socket, but may not
+   * have completed CONNECT/CONNACK.
+   * It is fairly safe to start queuing messages at this point, but if you
+   * want to be really sure you should wait until after a successful call to
+   * the connect callback.
+   * In this case we know it is 1 second before we start publishing.
+   */
+  while(running){
+    //publish_sensor_data(mosq);
```

Comment:
It would be interesting to add some PUBLISH example here. (Maybe in a future PR?).
For now please remove and call sleep to avoid wasting too many CPU cycles here.